### PR TITLE
Don't auto-replace PhasedXPowGate instances with XPowGate or YPowGate

### PR DESCRIPTION
**Pull request by Strilanc**

### CODE REVIEW

File: cirq/ops/phased_x_gate.py [hunk 1]
```diff
@@ -211,5 +189,20 @@ def _canonical_exponent(self):
...
 
+    def _value_equality_values_cls_(self):
+        if self.phase_exponent == 0:
+            return common_gates.XPowGate
+        if self.phase_exponent == 0.5:
+            return common_gates.YPowGate
+        return PhasedXPowGate
+
     def _value_equality_values_(self):
+        if self.phase_exponent == 0:
```
Comment (dabacon) [inline cirq/ops/phased_x_gate.py]: does this delegation really provide value?

**Comment by mpharrigan** [comment]
1) This would make my life a lot easier :) so big +1 from me
2) I'm also mildly confused about the value equality special casing. Could you motivate this a bit more?

**Comment by Strilanc** [comment]
I special cased the equality because not doing so broke many, many tests. I believe all those tests *could* be rewritten, but it was tricky to do so.

**Comment by dabacon** [comment]
Let's open an issue then for this then, since I think it seems strickly not necessary.

**Pull request merged**